interfaces/cython/cantera/ck2yaml.py

```diff
@@ -691,13 +691,14 @@ def __init__(self, label, geometry, well_depth, collision_diameter,
                  dipole_moment, polarizability, z_rot, note=''):
 
         try:
-            geometry = int(geometry)
-        except ValueError:
-            raise InputError(
-                "Bad geometry flag '{}' for species '{}', is the flag a float "
-                "or character? It should be an integer.", geometry, label)
-        if geometry not in (0, 1, 2):
-            raise InputError("Bad geometry flag '{}' for species '{}'.", geometry, label)
+            if float(geometry) == int(float(geometry)):
+                geometry = int(float(geometry))
+            else:
+                raise ValueError("Flag should be an integer.")
+            if geometry not in (0, 1, 2):
+                raise ValueError("Flag value should be 0, 1, or 2.")
```

Comment:
The old `InputError` should remain unchanged (outside of the `try` .. `except` block), as it will show the section where the error is raised.